src/Parsing/Impl/Ast/ScopeInfo.cs

```diff
@@ -1,41 +1,32 @@
-// Copyright(c) Microsoft Corporation
-// All rights reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the License); you may not use
-// this file except in compliance with the License. You may obtain a copy of the
-// License at http://www.apache.org/licenses/LICENSE-2.0
-//
-// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
-// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
-// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
-// MERCHANTABILITY OR NON-INFRINGEMENT.
-//
-// See the Apache Version 2.0 License for specific language governing
-// permissions and limitations under the License.
-
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using Microsoft.Python.Core;
+using Microsoft.Python.Parsing.Definition;
 
 namespace Microsoft.Python.Parsing.Ast {
-    public abstract class ScopeStatement : Statement {
```

Comment:
The parser's API shouldn't really be changing. Removing this entire type is going to cause issues for those who depend on our API (which is why its in its own assembly).

Reply:
Deprecating behavior or adding more bases/interface implementations should be fine, but the overall API itself needs some care to change.

Reply:
Ok, so would it be enough to add `ScopeStatement` back in, but delegate its existing public interfaces to `ScopeInfo`? The other parts are internal, so I assume that it is fine to move them out of the class. 

I can have `ScopeStatement` implement `IScopeStatement` and then `ClassDefinition` and `FunctionDefinition` extend `ScopeStatement`.

Reply:
I don't think FindScopes is going to be a problem, `ScopeStatement` implements `IScope` so people can still pass in their same objects. I also added the `ScopeStatement` class back and marked it as deprecated. The public fields should remain the same.

Reply:
Sorry, I don't mean to say to actually "deprecate" the ScopeStatement class. It's just an AST node.

Reply:
Ok I'll change it back then. I think that people in the future should not use ScopeStatement in their code and it should ultimately be phased out because it doesn't do much.